README.md

```diff
@@ -209,8 +120,258 @@ type Droid implements Character {
+
+Custom Date scalar type example:
+
+```php
+$dateType = newScalarType([
+    'name'         => 'Date',
+    'serialize'    => function (DateTime $value) {
+        return $value->format('Y-m-d');
+    },
+    'parseValue'   => function (DateTime $value) {
+        return $value->format('Y-m-d');
+    },
+    'parseLiteral' => function ($node) {
+        if ($node instanceof StringValueNode) {
+            return new DateTime($node->getValue());
+        }
+        return null;
+    },
+]);
+
+$schema = buildSchema($source, [
+    'Query' => QueryResolver::class,
+    [
+        'customTypes' => [$dateType],
+    ],
+]);
+```
+
+Every scalar has to be coerced, which is done by three different functions. The `serialize` function converts a 
+PHP value into the corresponding output value. The`parseValue` function converts a variable input value into the
+corresponding PHP value and the `parseLiteral` function converts an AST literal into the corresponding PHP value.
+
+### Advance usage
+
+If you are looking for something that isn't yet covered by this documentation your best bet is to take a look at the 
+[tests](./tests) in this project. You'll be surprised how many examples you'll find there.
+
+### Integration
+
+#### Laravel
+
+In Laravel you need to create a GraphQL service provider, a GraphQL service, a GraphQL controller and map a route for
+handling the GraphQL requests.
+
+**app\GraphQL\GraphQLServiceProvider.php**
+```php
+class GraphQLServiceProvider
+{
+    public function register()
+    {
+        $this->app->singleton(GraphQLService::class, function () {
+            $schemaDef = file_get_contents(__DIR__ . '/schema.graphqls');
+
+            $executableSchema = buildSchema($schemaDef, [
+                'Query' => QueryResolver::class,
+            ]);
+
+            return new GraphQLService($executableSchema);
+        });
+    }
+}
+```
+
+**app\GraphQL\GraphQLService.php**
+```php
+class GraphQLService
+{
+    private $schema;
+
+    public function __construct(Schema $schema)
+    {
+        $this->schema = $schema;
+    }
+
+    public function executeQuery(string $query, array $variables, ?string $operationName): array
+    {
+        return graphql($this->schema, $query, null, null, $variables, $operationName);
+    }
+}
+```
+
+**app\GraphQL\GraphQLController.php**
+```php
+class GraphQLController extends Controller
+{
+    private $graphqlService;
+
+    public function __construct(GraphQLService $graphqlService)
+    {
+        $this->graphqlService = $graphqlService;
+    }
+
+    public function handle(Request $request): JsonResponse
+    {
+        $query         = $request->get('query');
+        $variables     = $request->get('variables') ?? [];
+        $operationName = $request->get('operationName');
+
+        $result = $this->graphqlService->executeQuery($query, $variables, $operationName);
+
+        return response()->json($result);
+    }
+}
+```
+
+**routes/api.php**
```

Comment:
This uses forward slashes and the previous filenames use backslashes. Should they match?